Mathlib/Topology/Algebra/Algebra.lean

```diff
@@ -574,6 +574,53 @@ theorem Subalgebra.topologicalClosure_comap_homeomorph (s : Subalgebra R A) {B :
   rw [w]
   exact f'.preimage_closure _
 
+variable (R)
+
+open Subalgebra
+
+/-- The topological closure of the subalgebra generated by a single element. -/
+def Algebra.elemental (x : A) : Subalgebra R A :=
+  (Algebra.adjoin R ({x} : Set A)).topologicalClosure
+
+@[deprecated (since := "2024-11-05")] alias Algebra.elementalAlgebra := Algebra.elemental
+
+namespace Algebra.elemental
+
+@[aesop safe apply (rule_sets := [SetLike])]
+theorem self_mem (x : A) : x ∈ elemental R x :=
+  le_topologicalClosure _ <| self_mem_adjoin_singleton R x
+
+@[deprecated (since := "2024-11-05")] alias _root_.Algebra.self_mem_elementalAlgebra := self_mem
+
+theorem le_of_mem {x : A} {s : Subalgebra R A} (hx : x ∈ s)
+    (hs : IsClosed (s : Set A)) :
+    elemental R x ≤ s :=
+  topologicalClosure_minimal _ (adjoin_le <| by simpa using hx) hs
+
+instance isClosed (x : A) : IsClosed (elemental R x : Set A) :=
+  isClosed_topologicalClosure _
+
+instance [T2Space A] {x : A} : CommSemiring (elemental R x) :=
+  commSemiringTopologicalClosure _
+    letI : CommSemiring (adjoin R {x}) :=
+      adjoinCommSemiringOfComm R fun y hy z hz => by
+        rw [mem_singleton_iff] at hy hz
+        rw [hy, hz]
+    fun _ _ => mul_comm _ _
```

Comment:
I'm a bit scared by the fact that [Subalgebra.commSemiringTopologicalClosure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Algebra.html#Subalgebra.commSemiringTopologicalClosure) is not reducible (see e.g [Subalgebra.commRingTopologicalClosure](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Algebra.html#Subalgebra.commRingTopologicalClosure) for comparison). Would it make sense to explicitly override the `toSemiring` field to make sure all data is easily reducible to what it should be?

Reply:
oh, I think those should just be marked `abbrev` instead. I'll make that change later.

Reply:
I've done this in #18700 (along with other things)